File: Client/src/Student/model/student.ts
import { Identity } from "../../model/Identity";
import { Grades } from "./grades";
import { SubjectStudent } from "./subject-student";

export class Student extends Identity{
   Name :string ='';
   Phone : string ='';
   Gender : string='';
   LastName :string =''
   Age : number =0;
   Tutor : string='';
   TutorRelationShip :string =''
   TutorPhone :string =''
   SubjectStudentList : SubjectStudent[]=[];
   CreatedAt : Date = new Date();
   CreateBy : string ='';

/**
 *
 */
constructor(init?: Partial<Student>) {
    super();
    Object.assign(this, init);
  }

  static JsonInstance(entity: any): Student {
   var _object =   entity ? Object.assign(new Student(), entity): new Student();
    _object.SubjectStudentList = SubjectStudent.JsonInstaces(entity?.SubjectStudentList);
    return _object;

  }

  static JsonInstaces(entity: any[]): Student[] {
    return entity.map(item => Student.JsonInstance(item));
  }

}

File: Api/EscuelaPrimaria/Entity/Student.cs
using System.ComponentModel.DataAnnotations;

namespace EscuelaPrimaria.Entity
{
    public partial class Student : Entity
    {
     

        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string LastName { get; set; } = string.Empty;
        [Phone]
        [MaxLength(20)]
        public string? Phone { get; set; }
    
        public int? Age { get; set; }
        public string? Gender { get; set; }
        public string? Tutor { get; set; }
        public string TutorRelationShip { get; set; } = string.Empty;
        public string TutorPhone { get; set; } = string.Empty;
        [Phone]
        public bool? Active { get; set; } = true;

        public ICollection<SubjectStudent> SubjectStudents { get; set; } = new List<SubjectStudent>();
        public ICollection<Attendence> Attendences { get; set; } = new List<Attendence>();
        
    }
}


